modules/Bio/EnsEMBL/Production/Pipeline/Xrefs/Base.pm

```diff
@@ -83,17 +84,17 @@ sub download_file {
       if ( !match_glob( basename( $uri->path() ), $remote_file ) ) { next; }
       $remote_file =~ s/\///g;
       $file_path = catfile($dest_dir, basename($remote_file));
-      mkdir(dirname($file_path));
-      $ftp->get( $remote_file, $file_path );
+      $ftp->get( $remote_file, $file_path ) unless $skip_download_if_file_present and -f $file_path;
     }
   } elsif ($uri->scheme eq 'http' || $uri->scheme eq 'https') {
     $file_path = catfile($dest_dir, basename($uri->path));
-    mkdir(dirname($file_path));
-    open OUT, ">$file_path" or die "Couldn't open file $file_path $!";
-    my $http = HTTP::Tiny->new();
-    my $response = $http->get($uri->as_string());
-    print OUT $response->{content};
-    close OUT;
+    unless ($skip_download_if_file_present and -f $file_path) {
+      open OUT, ">$file_path" or die "Couldn't open file $file_path $!";
+      my $http = HTTP::Tiny->new();
+      my $response = $http->get($uri->as_string());
+      print OUT $response->{content};
```

Comment:
I realize this is just wrapping old, existing functionality in an unless block, but sucking the file in the memory then spitting it out like that just feels... dirty. HTTP::Tiny has a mirror() routine.

https://metacpan.org/pod/HTTP::Tiny#mirror